services/route_manager/src/main/java/com/futurewei/alcor/route/controller/RouterController.java

```diff
@@ -259,7 +333,7 @@ public RouteTableWebJson updateVpcRouteTable(@PathVariable String projectid, @Pa
     @DurationStatistics
     public RouteTablesWebJson getVpcRouteTables(@PathVariable String projectid, @PathVariable String vpcid) throws Exception {
 
-        List<RouteTable> routetables = new ArrayList<>();
+        List<RouteTable> routetables = null;
```

Comment:
@DavidLiu506 If you set this as `null` and this function cannot find vpcroutetables, line 299 in this code will get NPE (null point exception). 

Also, please uncomment line 309 in this file. We need to send `internalRouterInfo` to DPM.

Reply:
line send internalRouterInfo to DPM uncomment.

updateVpcRouteTable in RouterController will not get NPE because  updateVpcRouteTable in RouterServiceImpl already handle it.

Reply:
@DavidLiu506 I didn't see you have handled if this function return `null` in `updateVpcRouteTable`:

```
        List<RouteTable> vpcRouteTables = router.getVpcRouteTables();
        for (RouteTable vpcRouteTable : vpcRouteTables) {
            String routeTableType = vpcRouteTable.getRouteTableType();
            if (RouteTableType.VPC.getRouteTableType().equals(routeTableType)) {
                return vpcRouteTable;
            }
        }
```

Reply:
Sorry, you are right. Fixed.